roles/translator/src/status.rs

```diff
@@ -1,12 +1,44 @@
 use crate::error::Error;
 
 #[derive(Debug)]
-pub enum State {
-    Shutdown(Error),
+pub enum State<'a> {
+    Shutdown(Error<'a>),
     Healthy(String),
 }
 
 #[derive(Debug)]
-pub struct Status {
-    pub state: State,
+pub struct Status<'a> {
+    pub state: State<'a>,
+}
+
+/// # Description
+/// This macro handles errors inserting error handling logic for a given `Result<T, crate::error::Error<'a>>`
+/// it is used by passing in a `Sender<crate::status::Status>` as the first parameter
+/// and a `Result<T, crate::error::Error<'a>>` as the second parameter
+/// # Example
+/// ```
+/// let (tx_status: Sender<Status>, rx_status) = async_channel::unbounded();
+/// let variable = handle_result!(
+///     tx_status,
+///     type_.try_into()
+/// );
+/// ```
+#[macro_export]
+macro_rules! handle_result {
+    ($sender:expr, $res:expr) => {
+        match $res {
+            Ok(val) => val,
+            Err(e) => {
+                tracing::error!("Error: {:?}", e);
+                // send status
+                $sender
+                    .send($crate::status::Status {
+                        state: $crate::status::State::Shutdown(e.into()),
+                    })
+                    .await
+                    .unwrap();
+                panic!("Error");
```

Comment:
I'm not sold yet on this panic() - if we leave it it should have a big fat comment explaining it - something like "this panic is meant to 1.) escape an infinite loop inside a async task 2.) log the file/line of the panic

what happens when you panic inside a macro? How does it spit out a line number as the line numbers would be wonky right?

Reply:
the macro essentially injects the code at the source so this is the error I got 
![image](https://user-images.githubusercontent.com/105945999/208130953-d7e7c2e1-0812-475a-b43f-dd4ba7ee6e78.png)

I just put the panic() since I am not sure what we want to do yet. I think we can have a match statement inside the macro 
to handle each of the errors independently